### PR TITLE
Monitorbackend

### PR DESCRIPTION
**Pull request by aginzberg**
Added backend functionality to the requester task monitoring. Added alias field to worker model. Front end could still use some work and we should decide what additional fields if any we want to include on the monitoring page. Disabled the ability to click on individual worker tasks - unclear what purpose that serves right now.

### CODE REVIEW

**Comment by mbernst** [comment]
:+1: great to see you diving in!

**Comment by neilthemathguy** [comment]
+1

File: crowdsourcing/serializers/template.py [hunk 1]
```diff
@@ -10,8 +10,9 @@ class Meta:
 
 
 class TemplateItemSerializer(serializers.ModelSerializer):
-    class Meta:
-        model = models.TemplateItem
+	template = TemplateSerializer()
```
Comment (nistala) [inline crowdsourcing/serializers/template.py]: do we need this line?

File: crowdsourcing/serializers/worker.py [hunk 2]
```diff
@@ -125,14 +126,17 @@ def create(self, **kwargs):
         return worker_skill
 
 
-class TaskWorkerSerializer(serializers.ModelSerializer):
+class TaskWorkerSerializer (serializers.ModelSerializer):
+    worker = WorkerSerializer()
```
Comment (nistala) [inline crowdsourcing/serializers/worker.py]: is this used?

File: crowdsourcing/serializers/worker.py [hunk 3]
```diff
@@ -125,14 +126,17 @@ def create(self, **kwargs):
...
     class Meta:
         model = models.TaskWorker
         fields = ('task', 'worker', 'created_timestamp', 'last_updated')
         read_only_fields = ('task', 'worker', 'created_timestamp', 'last_updated')
 
 
-class TaskWorkerResultSerializer(serializers.ModelSerializer):
+class TaskWorkerResultSerializer (serializers.ModelSerializer):
+    task_worker = TaskWorkerSerializer()
+    template_item = TemplateItemSerializer()
```
Comment (nistala) [inline crowdsourcing/serializers/worker.py]: these fields used?

File: staticfiles/js/monitor/services/monitor.service.js [hunk 4]
```diff
@@ -22,44 +22,17 @@
...
-    /**
-    * @name addMonitor
-    * @desc Try to create a new monitor
-    * @returns {Promise}
-    * @memberOf crowdsource.monitor.services.Monitor
-    */
-    function getWorkers(id) {
-      /*
+    function getTaskWorkerResults(){
       return $http({
```
Comment (nistala) [inline staticfiles/js/monitor/services/monitor.service.js]: would recommend this instead for oauth validation:
```
var settings = {
  url: '/api/task-worker-result/',
  method: 'GET'
}
return HttpService.doRequest(settings);
```

File: staticfiles/js/monitor/services/monitor.service.js [hunk 5]
```diff
@@ -22,44 +22,17 @@
     * @desc The Factory to be returned
```
Comment (nistala) [inline staticfiles/js/monitor/services/monitor.service.js]: You should inject 'HttpService'

File: staticfiles/templates/task/monitor.html [hunk 6]
```diff
@@ -55,7 +52,7 @@ <h3 class="panel-title"><i class="fa fa-calendar-o"></i> &nbsp; Milestone 1</h3>
 </div>
 
 
-<div class="modal fade" id="myModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true">
+<!-- <div class="modal fade" id="myModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true">
```
Comment (nistala) [inline staticfiles/templates/task/monitor.html]: we have history with git, if you're not using it, it's fine to delete it.

File: staticfiles/templates/task/monitor.html [hunk 7]
```diff
@@ -15,12 +15,12 @@
```
Comment (neilthemathguy) [inline staticfiles/templates/task/monitor.html]: Please move this under Requester Dashboard - Monitor Tab

File: staticfiles/js/monitor/services/monitor.service.js (hunk 4 above)
Comment (dmorina) [inline staticfiles/js/monitor/services/monitor.service.js]: yeah this will be protected in the near future so it wont work

**Pull request merged**